app/models/rdv_context.rb

```diff
@@ -21,7 +21,8 @@ class RdvContext < ApplicationRecord
   }
   scope :attention_needed, -> { status(STATUSES_WITH_ATTENTION_NEEDED) }
   scope :invited_before_time_window, lambda { |number_of_days_before_action_required|
-    where.not(id: Invitation.sent_in_time_window(number_of_days_before_action_required).pluck(:rdv_context_id).uniq)
+    where(id: RdvContext.includes(:invitations).select(&:first_sent_invitation).map(&:id))
+      .where.not(id: Invitation.sent_in_time_window(number_of_days_before_action_required).pluck(:rdv_context_id).uniq)
```

Comment:
Je suis pas sûr de comprendre ton scope `RdvContext.includes(:invitations).select(&:first_sent_invitation).map(&:id)` 🤔  . 
En gros ici tu récupères les `rdv_contexts` qui ont des invitations envoyées comme dans une `:with_sent_invitations` sauf que c'est de manière bcp plus complexe vu que tu itères sur tous les contextes et toutes les invitations.